### PR TITLE
Make compatible with Gradle Config Cache

### LINKED ISSUE #85: Configuration Cache compatibility with Gradle 8.1+
When trying to use artifactregistry-gradle-plugin:2.2.0 with Configuartion Cache Gradle 8.1 it errors out with:

1 problem was found storing the configuration cache.
- Plugin 'com.google.cloud.artifactregistry.gradle-plugin': external process started 'gcloud config config-helper --format=json(credential)'
  See https://docs.gradle.org/8.1-rc-3/userguide/configuration_cache.html#config_cache:requirements:external_processes

### PR DESCRIPTION
**Pull request by bjoernmayer**
Fixes #85 

This is based upon @Naitbit solution.

I introduce `CommandExecutor` in order to abstract the command execution of `gcloud`.

The implementation of Maven uses the same strategy as before.
The implementation for Gradle makes uses the providerFactory.

### CODE REVIEW

**Review by aabdelhafez — APPROVED** [review]
Awesome work!

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.6.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
```
Comment (Naitbit) [inline gradle/wrapper/gradle-wrapper.properties:3]: At this point maybe it can be bumped all the way to 8.7 just in case there are any new warnings?

Comment (bjoernmayer) [reply gradle/wrapper/gradle-wrapper.properties]: I had issues building with 8.7. I just tried again.
However, I will probably move the gradle update related changes to a separate PR

**Comment by TSampley** [comment]
@bjoernmayer have you already signed the CLA? https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools/pull/92/checks?check_run_id=23209124250

I'm trying to figure out why Google hasn't merged this yet when it affects so many *modern* developers.

**Comment by bjoernmayer** [comment]
> @bjoernmayer have you already signed the CLA? https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools/pull/92/checks?check_run_id=23209124250
> 
> I'm trying to figure out why Google hasn't merged this yet when it affects so many _modern_ developers.

All signed. See the [build](https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools/pull/92/checks?check_run_id=24106167510)

I even mentioned the Devs in this [PR](https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools/pull/93).
No reaction whatsoever.
That's why I decided to setup [my own Plugin](https://github.com/bjoernmayer/artifactregistry-gradle).

**Comment by yihanzhen** [comment]
@bjoernmayer very sorry I might have ruined this PR by updating the branch (looks like you had more commits to your branch) and now the build failed. Any chance you can fix this?

**Comment by bjoernmayer** [comment]
> @bjoernmayer very sorry I might have ruined this PR by updating the branch (looks like you had more commits to your branch) and now the build failed. Any chance you can fix this?

Sure thing. 
Unfortunately, I do not know what is broken as I cannot see the build output.

**Comment by yihanzhen** [comment]
> > @bjoernmayer very sorry I might have ruined this PR by updating the branch (looks like you had more commits to your branch) and now the build failed. Any chance you can fix this?
> 
> Sure thing. Unfortunately, I do not know what is broken as I cannot see the build output.

So it seems a lot of changes were not originally in this PR. After pulling in the new changes, the error is:

```
/workspace/artifactregistry-gradle-plugin/src/main/java/com/google/cloud/artifactregistry/gradle/plugin/ProviderFactoryCommandExecutor.java:6: error: cannot find symbol
import org.gradle.process.ExecOutput;
                         ^
  symbol:   class ExecOutput
  location: package org.gradle.process
/workspace/artifactregistry-gradle-plugin/src/main/java/com/google/cloud/artifactregistry/gradle/plugin/ProviderFactoryCommandExecutor.java:27: error: cannot find symbol
        ExecOutput execOutput;
        ^
  symbol:   class ExecOutput
  location: class ProviderFactoryCommandExecutor
/workspace/artifactregistry-gradle-plugin/src/main/java/com/google/cloud/artifactregistry/gradle/plugin/ProviderFactoryCommandExecutor.java:29: error: cannot find symbol
            execOutput = providerFactory.exec(execSpec -> {
                                        ^
  symbol:   method exec((execSpec)[...]t); })
  location: variable providerFactory of type ProviderFactory
/workspace/artifactregistry-gradle-plugin/src/main/java/com/google/cloud/artifactregistry/gradle/plugin/ArtifactRegistryGradlePlugin.java:99: error: cannot find symbol
        providerFactory = ((Settings) o).getProviders();
                                        ^
  symbol:   method getProviders()
  location: interface Settings
4 errors
```

I think it might still be better to stick to the just the gradle update in the PR, if it still possible for you :)

**Comment by bjoernmayer** [comment]
@yihanzhen are we able to merge https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools/pull/93?

I could merge master then, which brings PR down to the config cache changes

**Comment by yihanzhen** [comment]
Yep, just merged #93

**Comment by bjoernmayer** [comment]
@yihanzhen please retrigger the build :)

**Comment by pauldthomson** [comment]
Any updates on this one?

**Review by yihanzhen — APPROVED** [review]

**Comment by pauldthomson** [comment]
@yihanzhen what's the release process for this? Is it automated?

**Comment by yihanzhen** [comment]
Hey sorry I was super busy this week. I'll cut a release next Monday

**Comment by pauldthomson** [comment]
Ah all good, I just wasn't sure if something needed to be done etc

Thank you 🙂

**Comment by yihanzhen** [comment]
All done :)

**Pull request merged**